src/vocabulary/mod.rs

```diff
@@ -84,24 +124,27 @@ impl Vocabulary {
             });
         };
         for (token, token_id) in tokenizer.get_vocab(false) {
-            let token_bytes = processor.process(token)?;
-            // TODO: lossy is temp:
-            // - in python in was handled by byte_symbol function
-            // - interface needs to be redefined to treat Token type as bytes: Vec<u8>
-            let processed_token = String::from_utf8_lossy(&token_bytes);
-            vocabulary = vocabulary.insert(processed_token, token_id);
+            if token_id != eos_token_id {
+                let processed_token = processor.process(&token)?;
+                vocabulary.try_insert(processed_token, token_id)?;
+            }
         }
 
         Ok(vocabulary)
     }
 
-    /// Per provided token returns vector of `TokenId`s if available in the vocabulary.
-    pub fn token_to_ids(&self, token: &str) -> Option<&Vec<TokenId>> {
-        self.tokens.get(token)
+    /// Returns all tokens with their token ids in vocabulary
+    pub fn tokens_to_ids(&self) -> &HashMap<Token, Vec<TokenId>> {
+        &self.tokens
+    }
+
+    /// Returns all token ids per provided token if available in the vocabulary.
+    pub fn token_to_ids(&self, token: impl AsRef<[u8]>) -> Option<&Vec<TokenId>> {
+        self.tokens.get(token.as_ref())
```

Comment:
I'm thinking about renaming this pair, it's proven to be very confusing while I was myself using it:
- `token_to_ids` -> `tokens`
- `tokens_to_ids` -> `token_ids`

@umut-sahin any other good options come to mind?

Reply:
`token_to_ids` is for retrieving the ids of a token, so renaming it to `tokens` doesn't make sense.
`tokens_to_ids` is for retrieving the full dict, so renaming it to `token_ids` doesn't make sense.

I think you meant the opposite :smile:
Which makes sense!

An alternative if you want, but either is fine :+1: 
- `vocabulary.token_to_ids(token)` -> `vocabulary.ids_of(token)`
- `vocabulary.tokens_to_ids()` -> `vocabulary.token_dict()`

Reply:
Lol yeah, the other way around, telling you - confusing! 😄 
Thanks for ideas, let's see!